QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,126 @@
+
+### Contributor Origin [2.ii]
+This package does not have a confirmation of contributor origin policy.
+
+### Peer Review Policy [2.iii]
+
+There is no stated peer review policy, but each pull request is at least reviewed by the core maintainer.
+
+### Continuous Integration [2.iv]
+
+Changes must pass CI on AMD 64 for:
+- Ubuntu 16.04 for gcc 4.9 and 5.5
+- MacOS with XCode 9.4 with Apple LLVM version 9.1.0
+- Windows with Microsoft Visual Studio 2015, MSVC 18.0
+
+ARM 64 support is not directly evaluated by this package's CI process.
+
+Regular CI results can be seen here:
+* [Ubuntu and OS X](https://travis-ci.org/github/foonathan/memory).
+* [Windows](https://ci.appveyor.com/project/foonathan/memory/branch/master).
+
+### Documentation Policy [2.v]
+
+This package does not have a stated documentation policy.
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+`foonathan_memory` has a [feature list](https://foonathan.net/memory/index.html) with descriptions of its main features.
+
+### Public API Documentation [3.ii]
+
+`foonathan_memory` has embedded API documentation that is generated using doxygen and is [hosted](https://foonathan.net/memory/index.html) alongside the feature documentation.
+There is documentation for nearly all of the public API.
+
+### License [3.iii]
+
+The license for `foonathan_memory` is zlib, a summary statement is provided in each source file, and a full copy of the license is in the [LICENSE](https://raw.githubusercontent.com/foonathan/memory/master/LICENSE) file.
+
+### Copyright Statements [3.iv]
+
+The copyright holders each provide a statement of copyright in each source code file in `foonathan_memory`.
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+Many features in `foonathan_memory` have corresponding tests which simulate typical usage, and they are located in the `test` directory.
+
+### Public API Testing [4.ii]
+
+Many parts of the public API have tests.
+
+### Coverage [4.iii]
+
+`foonathan_memory` does not track testing coverage.
+
+### Performance [4.iv]
+
+`foonathan_memory` does not conduct performance tests.
+
+### Linters and Static Analysis [4.v]
+
+`foonathan_memory` does not conduct tests with linters on its public CI.
+
+## Dependencies [5]
+
+`foonathan_memory` does not have any dependencies outside of the C++ standard library.
+
+## Platform Support [6]
+
+`foonathan_memory` supports the following platforms and tests each one:
+- Ubuntu 16.04 for gcc 4.9 and 5.5
+- MacOS with XCode 9.4 with Apple LLVM version 9.1.0
+- Windows with Microsoft Visual Studio 2015, MSVC 18.0
+
+ARM 64 support is not directly evaluated by this package's CI process.
+
+Regular CI results can be seen here:
+* [Ubuntu and OS X](https://travis-ci.org/github/foonathan/memory).
+* [Windows](https://ci.appveyor.com/project/foonathan/memory/branch/master).
+
+## Vulnerability Disclosure Policy [7.i]
+
+`foonathan_memory` does not have a Vulnerability Disclosure Policy
```

Comment:
Mind adding table with the summary of each of the topics? It will help external readers with the sections of the Quality Levels partially addressed. (For example, the API testing which is partially addressed)

Reply:
A summary of its current progress can be found in the related issue (https://github.com/eProsima/foonathan_memory_vendor/issues/23).